osu.Game.Rulesets.Osu/Objects/Drawables/DrawableHitCircle.cs

```diff
@@ -182,6 +182,8 @@ protected override void UpdateHitStateTransforms(ArmedState state)
             // todo: temporary / arbitrary, used for lifetime optimisation.
             this.Delay(800).FadeOut();
 
+            (CirclePiece.Drawable as IMainCirclePiece)?.Animate(state);
```

Comment:
You may be wondering how this helps in any way. Basically it changes the ordering of execution so the custom logic in `DrawableOsuEditRuleset` runs *after* this is applied. May seem like a hack, but if you look at this change as an isolated one I think it actually feels more elegant than binding to `ApplyCustomUpdateState` in `xCirclePiece`. Open to feedback and discussion, regardless.